flame/main.cpp

```diff
@@ -113,13 +113,19 @@ void parse_flowspec(std::string spec, std::queue<std::pair<uint64_t, uint64_t>>
         if (verbosity > 1) {
             std::cout << "adding QPS flow: " << nums[0] << "qps, " << nums[1] << "ms" << std::endl;
         }
-        result.push(std::make_pair(std::stol(nums[0]), std::stol(nums[1])));
+        long want_r = std::stol(nums[0]);
+        if (want_r < c_count) {
+            std::cerr << "WARNING: QPS flow limit is less than concurrent senders, changing limit to " << c_count << std::endl;
+            want_r = c_count;
+        }
+        result.push(std::make_pair(want_r, std::stol(nums[1])));
     }
 }
 
 void flow_change(std::queue<std::pair<uint64_t, uint64_t>> qps_flow,
-    std::shared_ptr<TokenBucket> rl,
-    int verbosity)
+    std::vector<std::shared_ptr<TokenBucket>> rl_list,
```

Comment:
Does it need to be a vector of `shared_ptr`s? It seems to me that the lifetime of each token bucket is the same.